akka-http-core/src/main/java/akka/http/javadsl/model/HttpMessage.java

```diff
@@ -72,6 +73,11 @@
         Self addHeaders(Iterable<HttpHeader> headers);
 
         /**
+         * Returns a copy of this message with the given http credential header added to the list of headers.
+         */
+        Self addCredentials(HttpCredentials credentials);
```

Comment:
That's a good addition 👍 
Since no-one else but us may extend `HttpMessage` we are allowed to add the mima filters to make this pass.

Failure of build was:

```
[info] akka-http-core: found 2 potential binary incompatibilities while checking against com.typesafe.akka:akka-http-core_2.11:2.4.2  (filtered 122)
[error]  * abstract method addCredentials(akka.http.javadsl.model.headers.HttpCredentials)java.lang.Object in interface akka.http.javadsl.model.HttpMessage#MessageTransformations is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.http.javadsl.model.HttpMessage#MessageTransformations.addCredentials")
[error]  * method addCredentials(akka.http.javadsl.model.headers.HttpCredentials)akka.http.scaladsl.model.HttpMessage in trait akka.http.scaladsl.model.HttpMessage is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.http.scaladsl.model.HttpMessage.addCredentials")
```

And yeah, we can add those filters, so please do so in MiMa.scala :)
Thanks a lot in advance!